docs/dev/architeture.md

```diff
@@ -0,0 +1,206 @@
+
+At present, the control process of `Flush` is a little complicated, so the details will be explained in another document.
+
+#### Compaction
+module path: https://github.com/CeresDB/ceresdb/tree/main/analytic_engine/src/compaction
+
+The data of `MemTable` is flushed as `SST`s, but the file size of recently flushed `SST` may be very small. And too small or too many `SST`s lead to the poor query performance. Therefore, `Compaction` is then introduced to rearrange the `SST`s so that the multiple smaller `SST` files can be compacted into a larger `SST` file.
+
+The detailed strategy of `Compaction` will also be described with `Flush` in subsequent documents.
+
+#### Manifest
+module path: https://github.com/CeresDB/ceresdb/tree/main/analytic_engine/src/meta
+
+`Manifest` takes responsibilities for managing metadata of `Analytic Engine` including:
+- Table schema and table options;
+- The sequence number where the newest flush finishes;
+- The information of `SST`, such as `SST` path.
+
+Now the `Manifest` is based on `WAL` (this is a different instance from the `WAL` mentioned above for newly written data) and in order to avoid infinite expansion of metadata (actually every `Flush` leads to an update on sst information), `Snapshot` is also introduced to clean up the history of metadata updates.
+
+#### Object Store
+module path: https://github.com/CeresDB/ceresdb/tree/main/components/object_store
+
+The `SST` generated by `Flush` needs to be persisted and the abstraction of the persistent storage device is `ObjectStore` including multiple implementations:
+- Based on local file system;
+- Based on [Alibaba Cloud OSS](https://www.alibabacloud.com/product/object-storage-service).
+
+The distributed architecture of CeresDB separates storage and computing, which requires `Object Store` needs to be a highly available and reliable service independent of CeresDB. Therefore, storage systems like [Alibaba Cloud OSS](https://www.alibabacloud.com/product/object-storage-service) is a good choice and in the future implementations on storage systems of some other cloud service providers is planned to provide.
+
+#### SST
+module path: https://github.com/CeresDB/ceresdb/tree/main/analytic_engine/src/sst
+
+Both `Flush` and `Compaction` involves `SST` and in the codebase `SST` itself is actually an abstraction that can have multiple specific implementations. The current implementation is based on [Parquet](https://parquet.apache.org/), which is a column-oriented data file format designed for efficient data storage and retrieval.
+
+The format of `SST` is very critical for retrieving data and is also the most important part to perform well in handling both timeseries and analytic workloads. At present, our [Parquet](https://parquet.apache.org/)-based implementation is good at processing analytic workload but is poor at processing timeseries workload. In our road map, a new hybrid storage format is planned to be able to process the two workloads at the same time.
```

Comment:
> In our road map, a new hybrid storage format is planned to be able to process the two workloads at the same time.

In our roadmap, we will explore more storage formats in order to achieve good performance in both workloads.